### ISSUE TITLE
unstack needs number of outputs as int arg

### ISSUE THREAD

**Issue by maxpumperla** [deeplearning4j/libnd4j#822]
guys, I’m integrating unstack into SameDiff right now. In libnd4j this op only has one int arg, namely dim (which is axis in TF). but the number of unstacked tensors num is something you should be able to configure as well. only if num is unspecified should it be inferred.

**Comment by raver119** [comment]
Max, it definitely non-configurable at my level. Even in TF num MUST be equal to specific dimension size.

**Comment by raver119** [comment]
```
x = tf.constant([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18], dtype=tf.float32, shape=(3, 2, 3))

z = tf.unstack(x, axis=0, name="zed", num=1)

with tf.Session() as sess:
    print z[0].eval().reshape(-1)
    print "shape: ", z[0].shape
    tf.train.write_graph(tf.Session().graph_def, "../models", "test_unstack.pb.txt", True)
```

**Comment by maxpumperla** [comment]
let's close this for now.
